File: Ejercicios Conexion BD/Rubi/mySQL.rb
require 'mysql2'

begin
  # Establece la conexión con la base de datos en XAMPP
  client = Mysql2::Client.new(
    host: "localhost",        # Dirección del servidor de la base de datos (local)
    username: "root",         # Usuario de MySQL
    password: "",             # Contraseña (debe configurarse si tiene una)
    database: "ecommerce3066478" # Base de datos específica
  )

  puts client
  puts "✅ Conexión exitosa"
  puts "🔎 Tipo de objeto de conexión: #{client.class}" # Muestra el tipo de objeto de conexión

  # Ejecutar consulta SQL para listar bases de datos
  results = client.query("SHOW DATABASES;")

  # Iterar sobre los resultados e imprimirlos
  results.each do |row|
    puts row
  end

rescue Mysql2::Error => e
  puts "❌ Error al conectar con MySQL: #{e.message}"

ensure
  # Cerrar la conexión si se estableció correctamente
  client.close if client
  puts "⚠️ Conexión cerrada"
end

File: Ejercicios Conexion/Go/mySQL.go


package main

import (
	"database/sql"
	"fmt"
	"log"

	_ "github.com/go-sql-driver/mysql"
)

func main() {
	// Configuración de conexión a MySQL
	dsn := "root:@tcp(localhost:3306)/ecommerce3066478"
	db, err := sql.Open("mysql", dsn)
	if err != nil {
		log.Fatalf("\u274C Error al conectar con MySQL: %v", err)
	}
	defer db.Close()

	// Verificar si la conexión es válida
	err = db.Ping()
	if err != nil {
		log.Fatalf("\u274C No se pudo establecer conexión: %v", err)
	}

	fmt.Println("\u2705 Conexión exitosa")
	fmt.Printf("\U0001F50E Tipo de objeto de conexión: %T\n", db)

	// Ejecutar la consulta para listar las bases de datos
	rows, err := db.Query("SHOW DATABASES;")
	if err != nil {
		log.Fatalf("\u274C Error al obtener bases de datos: %v", err)
	}
	defer rows.Close()

	// Imprimir bases de datos disponibles
	fmt.Println("\U0001F4C1 Bases de datos disponibles:")
	for rows.Next() {
		var dbName string
		if err := rows.Scan(&dbName); err != nil {
			log.Fatalf("\u274C Error al leer resultado: %v", err)
		}
		fmt.Printf("📌 %s\n", dbName)
	}

	// Verificar errores en la iteración
	if err := rows.Err(); err != nil {
		log.Fatalf("\u274C Error en la iteración: %v", err)
	}

	fmt.Println("\u2757 Conexión cerrada")
}
